### PR TITLE
Adding GitHub link exception for regenerated API docs

### PR DESCRIPTION
**Pull request by skrthomas**
Adding an exception for GitHub links in auto generated API documentation.

### CODE REVIEW

File: contributing_to_docs/doc_guidelines.adoc [hunk 1]
```diff
@@ -782,7 +782,7 @@ link:http://othersite.com/otherpath[friendly reference text]
 
 IMPORTANT: You must use `link:` before the start of the URL.
 
-IMPORTANT: You cannot link to a repository that is hosted on www.github.com.
+IMPORTANT: You cannot link to a repository that is hosted on www.github.com. An exception to this rule is when an autogenerated API reference contains a link to www.github.com.
```
Comment (bburt-rh) [inline contributing_to_docs/doc_guidelines.adoc:785]:
```suggestion
IMPORTANT: You cannot link to a repository that is hosted on www.github.com. An exception to this rule is when autogenerated API reference content contains a link to www.github.com.
```

Comment (abrennan89) [reply contributing_to_docs/doc_guidelines.adoc:785]: +1 for this change, thanks for opening the PR Sara!

Comment (skrthomas) [reply contributing_to_docs/doc_guidelines.adoc]: Agree. Thank you! Addressed :)

**Review by bburt-rh — APPROVED** [review]
Lgtm

**Review by abrennan89 — COMMENTED** [review]
/lgtm

**Review by jeana-redhat — APPROVED** [review]
LGTM

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

**Pull request merged**